src/adminCore/adminEntity.ts

```diff
@@ -1,11 +1,14 @@
-import { Connection, EntityMetadata, SelectQueryBuilder, Brackets } from '../utils/typeormProxy'
-import { EntityType } from '../types'
+import { EntityType, MaybePromise } from '../types'
 import { getDefaultWidget } from './widgets/utils'
 import DefaultAdminSite from './adminSite'
 import ManyToManyWidget from './widgets/manyToManyWidget'
 import { InvalidDisplayFieldsException } from './exceptions/invalidDisplayFields.exception'
 import { WidgetConstructor } from './widgets/widget.interface'
 
+import { Request, Response } from 'express'
+
+export type ListActionHandler = (request: Request, response: Response) => MaybePromise<void>
```

Comment:
Try to use multiple definitions instead of `MaybePromise`

Reply:
I investigated a few possibilities to replace the MaybePromise with multiple definitions without much success 
- I tried to imitate the method overloading properties : https://www.typescriptlang.org/docs/handbook/functions.html#overloads
- I tried using Generics to implement the overloading : https://www.typescriptlang.org/docs/handbook/generics.html
- I tried using an overloaded interface without success also : 
 `export interface ListActionHandler {
  (request: Request, response: Response): Promise<void>
  (request: Request, response: Response): void
}`

I created a ticket if anybody wants to keep investigating : https://trello.com/c/wIDwiJAW/281-aadev-refactor-the-definition-of-the-listactionhandler-and-changelistactionhandler-to-use-overloading-instead-of-maybepromise